x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/transform/transforms/TransformStats.java

```diff
@@ -151,7 +151,7 @@ public void writeTo(StreamOutput out) throws IOException {
         if (out.getVersion().onOrAfter(Version.V_7_4_0)) {
             out.writeString(id);
             // 7.13 introduced the waiting state, in older version report the state as started
-            if (out.getVersion().before(Version.V_8_0_0) && state.equals(State.WAITING)) {   // TODO: V_7_13_0
+            if (out.getVersion().before(Version.V_7_13_0) && state.equals(State.WAITING)) {
```

Comment:
Could we simplify this in `master`? IIUC `out.getVersion().before(Version.V_7_13_0)` will always be false.

Reply:
yes, that's correct. The reason why I defer the removal of BWC versioning is ease of backporting. We have tons of that code on master and I am looking forward to remove it. I would like to do this in 1 iteration after we close active transform development on 7.x.

If you look at the outer `if`, I have to remove the branching, which will result in a mess after backporting. Useful as well: As development happens on master, we would remove the "documentation" about BWC handling and potentially forget to handle it correctly as part of the backport.